app/services/task_service/gpt_generate_tests.rb

```diff
@@ -6,18 +6,38 @@ def initialize(task:)
       super()
       @task = task
       @message = task.description
-      @language = @task.programming_language.language || 'Python'
+      if @task.programming_language&.language
+        @language = @task.programming_language.language
+      else
+        raise MissingLanguageError
+      end
     end
 
     def execute
+      file_extensions = {'Python' => 'py', 'JavaScript' => 'js', 'Ruby' => 'rb', 'Java' => 'java'}
+      testing_frameworks = {'Python' => 'Pytest', 'JavaScript' => 'Jest', 'Ruby' => 'RSpec', 'Java' => 'JUnit'}
+
+      framework_name = testing_frameworks[@language]
+      framework_version = {'Pytest' => 6, 'Jest' => 27, 'RSpec' => 4, 'JUnit' => 5}
+      file_extension = file_extensions[@language]
+
       file_content = gpt_response
-      test_file = TaskFile.new(content: file_content, name: 'test.py', used_by_grader: true, visible: false, xml_id: SecureRandom.uuid)
-      testing_framework = TestingFramework.find_or_initialize_by(name: 'Pytest', version: 6)
-      test = Test.new(task: @task, title: 'AI-generated Test', xml_id: SecureRandom.uuid, files: [test_file], testing_framework:)
+      test_file = TaskFile.new(content: file_content, name: "test.#{file_extension}", used_by_grader: true, visible: false,
+        xml_id: SecureRandom.uuid)
+      testing_framework = TestingFramework.find_or_initialize_by(name: framework_name, version: framework_version[framework_name])
+      test = Test.new(task: @task, title: I18n.t('common.tests.generated_test'), xml_id: SecureRandom.uuid, files: [test_file],
```

Comment:
Probably, the translation currently located at `'common.tests.generated_test'` is better located at `'tests.model.generated_test'` (and `model.yml`), similarly to `tasks.model.copy_of_task`?

Reply:
I have moved the translation to the dedicated files as suggested. I did add an extra translation for 'invalid task description' in `tasks.yml` so I will not resolve this until that is also checked and confirmed to be correct. Thanks.